src/main/java/org/akaza/openclinica/controller/ScheduledExtractController.java

```diff
@@ -133,4 +159,14 @@ private ScheduledExtractJobDetailDTO convertEntityToDTO(ArchivedDatasetFileBean
         return scheduledExtractJobDetailDTO;
     }
 
+    private boolean userHasAccessToStudyOrSite(List<StudyUserRoleBean> userRoles, String studyOid) {
+        Study studyLevel = studyDao.findPublicStudy(studyOid);
+        studyOid = studyOid.trim().toUpperCase();
+        if (studyLevel != null) {
+            return validateService.isUserHasAccessToStudy(userRoles, studyOid);
+        } else {
+            return validateService.isUserHasAccessToSite(userRoles, studyOid);
```

Comment:
studyOid will not be a site oid. Given the study oid, you need to check if user has access to any of the sites in that study.

Reply:
Also, if "studyLevel" is null, it could mean that the given study oid is invalid. We can't assume that it is a site.